siad/API.md

```diff
@@ -0,0 +1,387 @@
+```
+Address should be an IP address of the form: "a.b.c.d:port"
+
+Response: standard
+
+#### /gateway/peer/remove
+
+Function: Will remove a peer from the gateway.
+
+Parameters:
+```
+struct {
+	Address string
+}
+```
+Address should be an IP address of the form: "a.b.c.d:port"
+
+Response: standard
+
+Host
+----
+
+Queries:
+
+* /host/announce
+* /host/config
+* /host/status
+
+#### /host/announce
+
+Function: The host will announce itself to the network as a source of storage.
+Generally only needs to be called once.
+
+Parameters: none
+
+Response: standard
+
+#### /host/config
+
+Function: Sets the configuration of the host.
+
+Parameters:
+```
+struct {
+	TotalStorage int
+	MinFilesize  int
+	MaxFilesize  int
+	MinDuration  int
+	MaxDuration  int
+	WindowSize   int
+	Price        int
+	Collateral   int
+}
+```
+TotalStorage (in bytes) is how much storage the host will rent to the network.
+
+MinFilesize is the minimum allowed file size.
+
+MaxFilesize is the maximum allowed file size.
+
+MinDuration is the minimum amount of time a contract is allowed to last.
+
+MaxDuration is the maximum amount of time a contract is allowed to last.
+
+WindowSize is the number of blocks a host has to prove they are holding the file.
+
+Price is the cost in Hastings per byte per block of hosting files on the network.
+
+Collateral is the amount of collateral the host will offer in Hastings per byte
+per block for losing files on the network.
+
+Response: standard
+
+#### /host/status
+
+Function: Queries the host for general information.
+
+Parameters: none
+
+Response:
+```
+struct {
+	TotalStorage     int
+	MinFilesize      int
+	MaxFilesize      int
+	MinDuration      int
+	MaxDuration      int
+	WindowSize       int
+	Price            int
+	Collateral       int
+	StorageRemaining int
+	NumContracts     int
+}
+```
+
+HostDB
+------
+
+Queries: none
+
+Miner
+-----
+
+Queries:
+
+* /miner/start
+* /miner/status
+* /miner/stop
+
+#### /miner/start
+
+Function: Starts the miner.
+
+Parameters: none
+
+Response: standard
+
+#### /miner/status
+
+Parameters: none
+
+Response:
+```
+struct {
+	Mining bool
+}
+```
+If the Mining flag is set, the miner is currently mining. Otherwise it is not.
+
+#### /miner/stop
+
+Function: Stops the miner.
+
+Parameters: none
+
+Response: standard
+
+Renter
+------
+
+Queries:
+
+* /renter/download
+* /renter/status
+* /renter/upload
+* /renter/uploadpath
+
+#### /renter/download
+
+Function: Starts a file download.
+
+Parameters:
+```
+struct {
+	Nickname    string
+	Destination string
+}
+```
+Nickname is the nickname of the file that has been uploaded to the network.
+
+Destination is the filepath that the file should be downloaded to.
+
+Response: standard
+
+#### /renter/status
+
+Function: Returns the status of the renter.
+
+Parameters: none
+
+Response:
+```
+struct {
+	Files []string
+}
+```
+Files is a list of all the files by their nickname.
+
+#### /renter/upload
+
+Function: Upload a file using a datastream.
```

Comment:
This section is special. The `Source` field is kind of misleading, because it isn't sent along with the `Nickname`, but rather as a "multipart file."

Hopefully this call will be removed soon anyway though.